Example/Tests/ViewModels/ConnectionsViewModelSpec.swift

```diff
@@ -0,0 +1,222 @@
+    var showSupportCall: Bool = false
+    var deleteConnectionCall: Bool = false
+    var reconnectCall: Bool = false
+    var consentsPressedCall: Bool = false
+    var updateViewsCall: Bool = false
+    var addPressedCall: Bool = false
+    var presentErrorCall: Bool = false
+    var showNoInternetConnectionAlertCall: Bool = false
+    var showDeleteConfirmationAlertCall: Bool = false
+    
+    func showEditConnectionAlert(placeholder: String, completion: @escaping (String) -> ()) {
+        showEditConnectionAlertCall = true
+    }
+    
+    func showSupport(email: String) {
+        showSupportCall = true
+    }
+    
+    func deleteConnection(completion: @escaping () -> ()) {
+        deleteConnectionCall = true
+    }
+    
+    func reconnect(by id: String) {
+        reconnectCall = true
+    }
+    
+    func consentsPressed(connectionId: String, consents: [SEConsentData]) {
+        consentsPressedCall = true
+    }
+    
+    func updateViews() {
+        updateViewsCall = true
+    }
+    
+    func addPressed() {
+        addPressedCall = true
+    }
+    
+    func presentError(_ error: String) {
+        presentErrorCall = true
+    }
+    
+    func showNoInternetConnectionAlert(completion: @escaping () -> Void) {
+        showNoInternetConnectionAlertCall = true
+    }
+    
+    func showDeleteConfirmationAlert(completion: @escaping () -> Void) {
+        showDeleteConfirmationAlertCall = true
+    }
+}
+
+final class ConnectionsViewModelSpec: BaseSpec {
+    
+    private var viewModel: ConnectionsViewModel!
+    private var networkSpy: ReachabilityManagerSpy!
+    
+    override func spec() {
+        beforeEach {
+            self.networkSpy = ReachabilityManagerSpy()
+            self.viewModel = ConnectionsViewModel(reachabilityManager: self.networkSpy)
+            ConnectionRepository.deleteAllConnections()
+        }
+        afterEach {
+            self.viewModel = nil
+            self.networkSpy = nil
+        }
+                
+        describe("count") {
+            it("should return count of connections") {
+                let connection = Connection()
+                connection.status = "active"
+
+                ConnectionRepository.save(connection)
+            
+                expect(self.viewModel.count).to(equal(1))
+            }
+        }
+        
+        describe("hasDataToShow") {
+            it("must display the data that needs to be shown") {
+                let connection = Connection()
+                connection.status = "active"
+
+                ConnectionRepository.save(connection)
+                
+                expect(self.viewModel.hasDataToShow).to(equal(true))
+            }
+            
+            it("will not display the data to be displayed") {
+                expect(self.viewModel.hasDataToShow).to(equal(false))
+            }
+        }
+        
+        describe("emptyViewData") {
+            context("when there is no connections") {
```

Comment:
We should use `context` when you have to test several cases. Here you have one case, so remove `context` and use only `it`